src/main/java/org/prebid/server/bidder/preciso/PrecisoBidder.java

```diff
@@ -0,0 +1,168 @@
+    private final JacksonMapper mapper;
+    private final CurrencyConversionService currencyConversionService;
+
+    public PrecisoBidder(String endpointUrl,
+                         CurrencyConversionService currencyConversionService,
+                         JacksonMapper mapper) {
+
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.currencyConversionService = Objects.requireNonNull(currencyConversionService);
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        final List<Imp> modifiedImps = new ArrayList<>();
+        final List<BidderError> errors = new ArrayList<>();
+
+        for (Imp imp : bidRequest.getImp()) {
+            try {
+                final ExtImpPreciso impExt = parseImpExt(imp);
+                final Price bidFloorPrice = resolveBidFloor(imp, impExt, bidRequest);
+
+                modifiedImps.add(modifyImp(imp, bidFloorPrice));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        if (errors.size() > 0) {
+            return Result.withErrors(errors);
+        }
+
+        final BidRequest outgoingRequest = bidRequest.toBuilder()
+                .cur(Collections.singletonList(BIDDER_CURRENCY)).imp(modifiedImps).build();
+
+        return Result.withValue(BidderUtil.defaultRequest(outgoingRequest, endpointUrl, mapper));
+
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.withValues(extractBids(bidResponse));
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private static List<BidderBid> extractBids(BidResponse bidResponse) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .filter(Objects::nonNull)
+                .map(bid -> BidderBid.of(bid, getBidType(bid), bidResponse.getCur()))
+                .toList();
+    }
+
+    private ExtImpPreciso parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), PRECISO_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private static Imp modifyImp(Imp imp, Price bidFloorPrice) {
```

Comment:
Please, place methods in call order. 
PS. For example, this method is called from `makeHttpRequests`, so it should be placed after it.